### PR TITLE
r/glue_schema - new resource

### LINKED ISSUE #16342: AWS Glue Schema Registry
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The AWS Glue Schema Registry is a new feature that allows you to centrally discover, control, and evolve data stream schemas.

### New or Affected Resource(s)

* `aws_glue_registry`
* `aws_glue_schema`

### Potential Terraform Configuration

```hcl
resource "aws_glue_registry" "example" {
  name = "example-registry"
}

resource "aws_glue_schema" "example" {
  registry_id {
    registry_arn = aws_glue_registry.example.arn
  }
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/11/control-evolution-data-streams-using-aws-glue-schema-registry/).
[Developer Guide](https://docs.aws.amazon.com/glue/latest/dg/schema-registry.html).

Requires [AWS SDK v1.35.32](https://github.com/aws/aws-sdk-go/releases/tag/v1.35.32):

* https://github.com/hashicorp/terraform-provider-aws/pull/16339

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #16342

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_glue_schema - add new resource
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSGlueSchema_'
--- PASS: TestAccAWSGlueSchema_disappears_registry (54.12s)
--- PASS: TestAccAWSGlueSchema_disappears (64.55s)
--- PASS: TestAccAWSGlueSchema_basic (83.05s)
--- PASS: TestAccAWSGlueSchema_compatibility (120.69s)
--- PASS: TestAccAWSGlueSchema_description (121.58s)
--- PASS: TestAccAWSGlueSchema_schemaDefUpdated (121.72s)
--- PASS: TestAccAWSGlueSchema_tags (158.50s)
```

### CODE REVIEW

File: aws/resource_aws_glue_schema.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+		input.Description = aws.String(d.Get("description").(string))
+		update = true
+	}
+
+	if d.HasChange("compatibility") {
+		input.Compatibility = aws.String(d.Get("compatibility").(string))
+		update = true
+	}
+
+	if update {
```
Comment (ewbankkit) [inline aws/resource_aws_glue_schema.go:202]: Could use [`d.HasChangesExcept("schema_definition", "tags")`](https://godoc.org/github.com/hashicorp/terraform-plugin-sdk/helper/schema#ResourceData.HasChangesExcept) here.

File: website/docs/r/glue_schema.html.markdown [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+* `compatibility` - (Required) The compatibility mode of the schema. Values values are: `NONE`, `DISABLED`, `BACKWARD`, `BACKWARD_ALL`, `FORWARD`, `FORWARD_ALL`, `FULL`, and `FULL_ALL`.
+* `schema_definition` - (Required) The schema definition using the `data_format` setting for `schema_name`.
+* `description` – (Optional) A description of the schema.
+* `tags` - (Optional) Key-value map of resource tags
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `arn` - Amazon Resource Name (ARN) of Glue Schema.
```
Comment (ewbankkit) [inline website/docs/r/glue_schema.html.markdown:41]:
```suggestion
* `arn` - Amazon Resource Name (ARN) of the schema.
```

File: website/docs/r/glue_schema.html.markdown [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+* `id` - Amazon Resource Name (ARN) of Glue Schema.
```
Comment (ewbankkit) [inline website/docs/r/glue_schema.html.markdown:42]:
```suggestion
* `id` - Amazon Resource Name (ARN) of the schema.
```

File: website/docs/r/glue_schema.html.markdown [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+* `registry_name` - The name of the registry.
```
Comment (ewbankkit) [inline website/docs/r/glue_schema.html.markdown:43]:
```suggestion
* `registry_name` - The name of the Glue Registry.
```

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSGlueSchema_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueSchema_ -timeout 120m
=== RUN   TestAccAWSGlueSchema_basic
=== PAUSE TestAccAWSGlueSchema_basic
=== RUN   TestAccAWSGlueSchema_description
=== PAUSE TestAccAWSGlueSchema_description
=== RUN   TestAccAWSGlueSchema_compatibility
=== PAUSE TestAccAWSGlueSchema_compatibility
=== RUN   TestAccAWSGlueSchema_tags
=== PAUSE TestAccAWSGlueSchema_tags
=== RUN   TestAccAWSGlueSchema_schemaDefUpdated
=== PAUSE TestAccAWSGlueSchema_schemaDefUpdated
=== RUN   TestAccAWSGlueSchema_disappears
=== PAUSE TestAccAWSGlueSchema_disappears
=== RUN   TestAccAWSGlueSchema_disappears_registry
=== PAUSE TestAccAWSGlueSchema_disappears_registry
=== CONT  TestAccAWSGlueSchema_basic
=== CONT  TestAccAWSGlueSchema_schemaDefUpdated
=== CONT  TestAccAWSGlueSchema_disappears_registry
=== CONT  TestAccAWSGlueSchema_disappears
=== CONT  TestAccAWSGlueSchema_compatibility
=== CONT  TestAccAWSGlueSchema_tags
=== CONT  TestAccAWSGlueSchema_description
--- PASS: TestAccAWSGlueSchema_disappears_registry (22.34s)
--- PASS: TestAccAWSGlueSchema_disappears (25.32s)
--- PASS: TestAccAWSGlueSchema_basic (29.83s)
--- PASS: TestAccAWSGlueSchema_compatibility (42.63s)
--- PASS: TestAccAWSGlueSchema_description (46.39s)
--- PASS: TestAccAWSGlueSchema_schemaDefUpdated (46.46s)
--- PASS: TestAccAWSGlueSchema_tags (57.64s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	57.752s
```

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @DrFaust92 

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSGlueSchema_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueSchema_ -timeout 120m
=== RUN   TestAccAWSGlueSchema_basic
=== PAUSE TestAccAWSGlueSchema_basic
=== RUN   TestAccAWSGlueSchema_description
=== PAUSE TestAccAWSGlueSchema_description
=== RUN   TestAccAWSGlueSchema_compatibility
=== PAUSE TestAccAWSGlueSchema_compatibility
=== RUN   TestAccAWSGlueSchema_tags
=== PAUSE TestAccAWSGlueSchema_tags
=== RUN   TestAccAWSGlueSchema_schemaDefUpdated
=== PAUSE TestAccAWSGlueSchema_schemaDefUpdated
=== RUN   TestAccAWSGlueSchema_disappears
=== PAUSE TestAccAWSGlueSchema_disappears
=== RUN   TestAccAWSGlueSchema_disappears_registry
=== PAUSE TestAccAWSGlueSchema_disappears_registry
=== CONT  TestAccAWSGlueSchema_basic
=== CONT  TestAccAWSGlueSchema_schemaDefUpdated
=== CONT  TestAccAWSGlueSchema_disappears
=== CONT  TestAccAWSGlueSchema_compatibility
=== CONT  TestAccAWSGlueSchema_disappears_registry
=== CONT  TestAccAWSGlueSchema_tags
=== CONT  TestAccAWSGlueSchema_description
--- PASS: TestAccAWSGlueSchema_disappears_registry (13.84s)
2020/12/08 10:09:19 [INFO] AWS Auth provider used: "SharedCredentialsProvider"
--- PASS: TestAccAWSGlueSchema_disappears (16.33s)
--- PASS: TestAccAWSGlueSchema_basic (20.30s)
--- PASS: TestAccAWSGlueSchema_compatibility (31.62s)
--- PASS: TestAccAWSGlueSchema_description (31.63s)
--- PASS: TestAccAWSGlueSchema_schemaDefUpdated (34.42s)
--- PASS: TestAccAWSGlueSchema_tags (42.69s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       44.370s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

Skipping as Registry/Schema are not available in GovCloud

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSGlueSchema_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueSchema_ -timeout 120m
=== RUN   TestAccAWSGlueSchema_basic
=== PAUSE TestAccAWSGlueSchema_basic
=== RUN   TestAccAWSGlueSchema_description
=== PAUSE TestAccAWSGlueSchema_description
=== RUN   TestAccAWSGlueSchema_compatibility
=== PAUSE TestAccAWSGlueSchema_compatibility
=== RUN   TestAccAWSGlueSchema_tags
=== PAUSE TestAccAWSGlueSchema_tags
=== RUN   TestAccAWSGlueSchema_schemaDefUpdated
=== PAUSE TestAccAWSGlueSchema_schemaDefUpdated
=== RUN   TestAccAWSGlueSchema_disappears
=== PAUSE TestAccAWSGlueSchema_disappears
=== RUN   TestAccAWSGlueSchema_disappears_registry
=== PAUSE TestAccAWSGlueSchema_disappears_registry
=== CONT  TestAccAWSGlueSchema_basic
=== CONT  TestAccAWSGlueSchema_schemaDefUpdated
=== CONT  TestAccAWSGlueSchema_compatibility
=== CONT  TestAccAWSGlueSchema_disappears_registry
=== CONT  TestAccAWSGlueSchema_description
=== CONT  TestAccAWSGlueSchema_disappears
=== CONT  TestAccAWSGlueSchema_tags
    resource_aws_glue_schema_test.go:295: skipping acceptance testing: AccessDeniedException: Operation not allowed
--- SKIP: TestAccAWSGlueSchema_tags (1.16s)
=== CONT  TestAccAWSGlueSchema_schemaDefUpdated
    resource_aws_glue_schema_test.go:295: skipping acceptance testing: AccessDeniedException: Operation not allowed
--- SKIP: TestAccAWSGlueSchema_schemaDefUpdated (1.16s)
=== CONT  TestAccAWSGlueSchema_description
    resource_aws_glue_schema_test.go:295: skipping acceptance testing: AccessDeniedException: Operation not allowed
--- SKIP: TestAccAWSGlueSchema_description (1.17s)
=== CONT  TestAccAWSGlueSchema_basic
    resource_aws_glue_schema_test.go:295: skipping acceptance testing: AccessDeniedException: Operation not allowed
--- SKIP: TestAccAWSGlueSchema_basic (1.18s)
=== CONT  TestAccAWSGlueSchema_disappears_registry
    resource_aws_glue_schema_test.go:295: skipping acceptance testing: AccessDeniedException: Operation not allowed
--- SKIP: TestAccAWSGlueSchema_disappears_registry (1.19s)
=== CONT  TestAccAWSGlueSchema_compatibility
    resource_aws_glue_schema_test.go:295: skipping acceptance testing: AccessDeniedException: Operation not allowed
--- SKIP: TestAccAWSGlueSchema_compatibility (1.19s)
=== CONT  TestAccAWSGlueSchema_disappears
    resource_aws_glue_schema_test.go:295: skipping acceptance testing: AccessDeniedException: Operation not allowed
--- SKIP: TestAccAWSGlueSchema_disappears (1.19s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       2.833s
```

**Pull request merged**